pkg/processor/runtime/rpc/abstract.go

```diff
@@ -453,3 +464,17 @@ func (r *AbstractRuntime) watchWrapperProcess() {
 
 	panic(fmt.Sprintf("Wrapper process for worker %d exited unexpectedly with: %s", r.Context.WorkerID, panicMessage))
 }
+
+// waitForProcessTermination will best effort wait few seconds to stop channel, if timeout - assume closed
+func (r *AbstractRuntime) waitForProcessTermination() {
```

Comment:
why not just use `time.After(...)` rather than this context thing?
Also, I'd log in each condition to help aid future debugging as needed